sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -948,7 +968,26 @@ public ServiceBusReceiverAsyncClient buildAsyncClient() {
          *     queueName()} or {@link #topicName(String) topicName()}, respectively.
          */
         public ServiceBusReceiverClient buildClient() {
-            return new ServiceBusReceiverClient(buildAsyncClient(), retryOptions.getTryTimeout());
+            return new ServiceBusReceiverClient(buildAsyncClient(false), retryOptions.getTryTimeout());
+        }
+
+        ServiceBusReceiverAsyncClient buildAsyncClient(boolean isAutoCompleteAllowed) {
+            final MessagingEntityType entityType = validateEntityPaths(logger, connectionStringEntityName, topicName,
+                queueName);
+            final String entityPath = getEntityPath(logger, entityType, queueName, topicName, subscriptionName,
+                subQueue);
+            validateAndThrow(prefetchCount);
+
+            if (!isAutoCompleteAllowed && enableAutoComplete) {
+                throw new IllegalStateException("'enableAutoComplete' is not support in synchronous client.");
```

Comment:
I thought that we are also doping this for sync clients ?

Reply:
Yes. It's for the callback model. The pull model will not have that.